jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/settings/component/binder/AbstractGridSettingsBinder.java

```diff
@@ -104,6 +111,16 @@ public void applyDataLoadingSettings(V component, S settings) {
         }
     }
 
+    protected List<? extends Grid.Column<?>> getApplicableColumns(V component) {
+        List<? extends Grid.Column<?>> componentColumns = getOrderedColumns(component);
+        List<? extends Grid.Column<?>> applicableColumns = componentColumns.stream()
+                .filter(c -> Objects.nonNull(c.getKey())).toList();
+        if (componentColumns.size() != applicableColumns.size()) {
+            log.debug("Grid has column without key specified, settings for it would not be stored");
```

Comment:
... one or more columns ..., ... for them won't be stored

Reply:
changed to null key lookup

Reply:
I didn't mean code changes. You iterate over column anyway and even with this implementation you still can have several columns with no `id`, so why not use a more general message? Also, log message must be **warning** as I mentioned before, so devs notice it.

One more thing, we prefer not hard code component name, since it can be changed + there are `Grid` and `TreeGrid` components. `component.getClass().getSimpleName()` would be better.

Reply:
warn will polute the logs in case the developer is happy with columns without key

Reply:
added component.getClass().getSimpleName()

Reply:
> warn will polute the logs in case the developer is happy with columns without key

that is the point
1. setting won't be applied if actual columns number isn't equal to number of columns in saved settings. So, with your current implementation, adding settings facet makes no sense, hence we need to **warn** devs, so it's easier for them to find the problem
2. if a dev adds settings facet, they assume that settings for all columns will be saved.

> added component.getClass().getSimpleName()
You still log message about a single column. I've already mentioned that there may be several and we need to use more general message.

Reply:
changed to warn, and log message, also they are currently not compared by size but there lookup for null-keyed column is performed